packages/neon_framework/lib/src/widgets/image.dart

```diff
@@ -336,70 +308,90 @@ class NeonUrlImage extends StatelessWidget {
   /// This can also be a data URI.
   final Uri uri;
 
-  /// Custom cache reviver function.
-  final CacheReviver? reviver;
-
-  /// Custom cache writer function.
-  final CacheWriter? writeCache;
-
-  /// {@macro NeonCachedImage.svgHint}
+  /// {@macro NeonImage.svgHint}
   final bool isSvgHint;
 
-  /// {@macro NeonCachedImage.size}
+  /// {@macro NeonImage.size}
   final Size? size;
 
-  /// {@macro NeonCachedImage.fit}
+  /// {@macro NeonImage.fit}
   final BoxFit? fit;
 
-  /// {@macro NeonCachedImage.svgColorFilter}
+  /// {@macro NeonImage.svgColorFilter}
   final ColorFilter? svgColorFilter;
 
-  /// {@macro NeonCachedImage.errorBuilder}
+  /// {@macro NeonImage.errorBuilder}
   final ErrorWidgetBuilder? errorBuilder;
 
   @override
-  Widget build(BuildContext context) {
-    final account = this.account ?? NeonProvider.of<AccountsBloc>(context).activeAccount.value!;
+  State<NeonUriImage> createState() => _NeonUriImageState();
+}
+
+class _NeonUriImageState extends State<NeonUriImage> {
+  late Account account;
+  final image = BehaviorSubject<Result<Uint8List>>();
+
+  @override
+  void initState() {
+    super.initState();
+
+    account = widget.account ?? NeonProvider.of<AccountsBloc>(context).activeAccount.value!;
 
-    final dataUri = uri.data;
+    unawaited(load());
+  }
+
+  @override
+  void dispose() {
+    unawaited(image.close());
 
-    return NeonCachedImage(
-      getImage: () async {
-        if (dataUri != null) {
-          return dataUri.contentAsBytes();
-        }
+    super.dispose();
+  }
 
-        final completedUri = account.completeUri(uri);
+  Future<void> load() async {
+    if (widget.uri.data != null) {
+      var data = widget.uri.data!.contentAsBytes();
+      try {
+        data = utf8.encode(ImageUtils.rewriteSvgDimensions(utf8.decode(data)));
```

Comment:
We must cut down the parsing that is going on.
We retrieve the data from cache as base64, convert it to a String (utf8) convert it back to binary (utf8) just so the SvgAssetLoader can convert it back into a String by decoding it (utf8). When rendering we also utf8 decode it to check whether it even is a Svg or a PNG/JPEG

Maybe the svg dimension rewrite can happen in the asset loader? Or maybe we should use a String eliminating the utf8 decoding for checking whether it is a PNG.